src/common.h

```diff
@@ -24,17 +24,30 @@ struct bpf_map_def SEC("maps/offsets") offsets = {
         .mono_ns = mono_ns,                    \
     }
 
+#define SET_OFFSET(CRC)                                   \
+    offset = CRC;                                         \
+    offset = (u64)bpf_map_lookup_elem(&offsets, &offset); \
+    if (!offset)                                          \
+        goto Skip;
+
 static __always_inline int read_value(void *base, u64 offset, void *dest, size_t dest_size)
 {
     /* null check the base pointer first */
     if (!base)
+    {
+        bpf_printk("Base should not be NULL\n");
         return -1;
+    }
 
     u64 _offset = (u64)bpf_map_lookup_elem(&offsets, &offset);
     if (_offset)
     {
-        return bpf_probe_read(dest, dest_size, base + *(u32 *)_offset);
+        int ret = bpf_probe_read(dest, dest_size, base + *(u32 *)_offset);
```

Comment:
did you mean to commit these changes?

Reply:
I have removed this and some of other changes that I think slipped through by accident. Let me know if there are other changes you want me to do while Dave is out.

Reply:
The last thing would be to remove the `printk`s. After that, I think the code changes look good. I'll do some testing to see if any kernel versions have issues with these new offsets.

Reply:
Just pushed up a change to remove them.